src/group_theory/category.lean

```diff
@@ -9,59 +9,51 @@ Copied from monoids.lean.
 -/
```

Comment:
This doesn't need to happen in this PR, but we need to rename/move this file, to be consistent with other instances. I'd suggest we move it to `algebra/Group/basic.lean`.

Reply:
I assumed that this PR is already large enough, so I shouldn't touch anything apart from migrating to the new concrete category API.